satellite/metabase/metadata.go

```diff
@@ -9,50 +9,45 @@ import (
 	"storj.io/common/storj"
 )
 
-// SetObjectMetadataLatestVersion contains arguments necessary for replacing an object metadata.
-type SetObjectMetadataLatestVersion struct {
-	ObjectLocation
+// UpdateObjectMetadata contains arguments necessary for replacing an object metadata.
+type UpdateObjectMetadata struct {
+	ObjectStream
 
 	EncryptedMetadata             []byte
 	EncryptedMetadataNonce        []byte
 	EncryptedMetadataEncryptedKey []byte
 }
 
-// SetObjectMetadataLatestVersion replaces an object metadata.
-func (db *DB) SetObjectMetadataLatestVersion(ctx context.Context, opts SetObjectMetadataLatestVersion) (err error) {
+// UpdateObjectMetadata updates an object metadata.
+func (db *DB) UpdateObjectMetadata(ctx context.Context, opts UpdateObjectMetadata) (err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	if err := opts.ObjectLocation.Verify(); err != nil {
+	if err := opts.ObjectStream.Verify(); err != nil {
 		return err
 	}
 
+	if opts.ObjectStream.Version <= 0 {
+		return ErrInvalidRequest.New("Version invalid: %v", opts.Version)
+	}
+
 	// TODO So the issue is that during a multipart upload of an object,
 	// uplink can update object metadata. If we add the arguments EncryptedMetadata
 	// to CommitObject, they will need to account for them being optional.
```

Comment:
This comment makes no sense to me because in fact EncryptedMetadata is an argument to db.CommitObject

Reply:
Here we restore the original `UpdateObjectMetadata` method. This comment was added by @mniewrzal who implemented it. Perhaps, the comment is already outdated, but Michal is not available at the moment to confirm. Thus, I'd prefer to keep it for now.